### DISCUSSION TITLE
export projects to csv

### DISCUSSION THREAD

**Discussion by letusskate** [community/community#47733] [Projects and Issues] [answered]
### Product Feedback
Once i use github projects, I got a feeling of smooth and natural. It's a really good thing for me to manage my coding plan.
but when I want to export it to make pictures or powerpoint, I found it really difficult. I have to copy and paste for about a whole day. I think if there is a icon to export the project to csv, it does make sence.

**Comment by jge162** [comment]
Hello @letusskate You could probably write a python script to manage that for you. Hopefully you know how to run a python script.
Should be easy. the possibilities are endless, I have even created a GitHub Action that could run a script (not this example) to commit the csv to a repo too. So it can be done yes. 

```python
import csv
import requests

# Replace the URL below with your GitHub project API URL
url = "https://api.github.com/repos/{your-username}/{your-repo}/issues"

# Replace the token below with your GitHub API token
token = "{your-github-api-token}"

headers = {"Authorization": f"Token {token}"}

response = requests.get(url, headers=headers)
issues = response.json()

# Replace the filename below with the name you want to give your CSV file
filename = "github_project_issues.csv"

with open(filename, "w", newline="") as csvfile:
    writer = csv.writer(csvfile)
    writer.writerow(["Title", "Body", "Labels"])

    for issue in issues:
        title = issue["title"]
        body = issue["body"]
        labels = [label["name"] for label in issue["labels"]]
        writer.writerow([title, body, ", ".join(labels)])

print(f"Data exported to {filename}")
```

**Reply by letusskate** [reply]
Thanks, it's helpful for issues' export, but I have many records in projects which aren't linked to issue, those would be missed. Also there are some columns which can be customized in projects won't appear in issue.

**Reply by southunitedraza** [reply]
> Hello @letusskate You could probably write a python script to manage that for you. Hopefully you know how to run a python script.
> Should be easy. the possibilities are endless, I have even created a GitHub Action that could run a script (not this example) to commit the csv to a repo too. So it can be done yes. 
> 
> ```python
> import csv
> import requests
> 
> # Replace the URL below with your GitHub project API URL
> url = "https://api.github.com/repos/{your-username}/{your-repo}/issues"
> 
> # Replace the token below with your GitHub API token
> token = "{your-github-api-token}"
> 
> headers = {"Authorization": f"Token {token}"}
> 
> response = requests.get(url, headers=headers)
> issues = response.json()
> 
> # Replace the filename below with the name you want to give your CSV file
> filename = "github_project_issues.csv"
> 
> with open(filename, "w", newline="") as csvfile:
>     writer = csv.writer(csvfile)
>     writer.writerow(["Title", "Body", "Labels"])
> 
>     for issue in issues:
>         title = issue["title"]
>         body = issue["body"]
>         labels = [label["name"] for label in issue["labels"]]
>         writer.writerow([title, body, ", ".join(labels)])
> 
> print(f"Data exported to {filename}")
> ```

[]()@@southunitedraza1313@gmail.com

**Reply by Farristom** [reply]
I need support

**Reply by sameer1098alam** [reply]
Still you need suport

**Reply by sakil2233** [reply]
47733

**Reply by nattchantasit-hash** [reply]
Some one is using GitHub to spy on me...Iam apinan sangthong

**Reply by ARCH-angel93** [reply]
Those are likely Server or Port Holders and Identified Users or Primary Contact for Server or Port Service based on personal or professional usage, as in they may not recognize when there are other users present or connected using the same connection contact methods, as if all on the same timezone or region or location based as a public node like point of reference, that could or may be not identifying based on specified exclusions from one to the other setup. It would be of matches of singularity rather than public multi user based shared text typed language written setup configurations in the codes or snips or how ever it is made or generated.

**Comment by AyanSinhaMahapatra** [comment]
This would be a nice to have feature. +1 for export to CSV

**Reply by pombredanne** [reply]
What about the import of a CSV?

**Comment by pombredanne** [comment]
@jge162 Your script is for plain issues, not for projects AFAICS?

**Reply by jge162** [reply]
The script is just a pseudo code idea. I do not know enough to write one for his issue. Just giving him/her ideas.

**Reply by Jhonn-tradintg01** [reply]
Jhonn

**Comment by pombredanne** [comment]
Short of an easy import/export from/to a tabular format (CSV or XLSX), this project feature is mostly harmless :]

**Comment by queenofcorgis** [comment]
👋🏼 Hi there! Thanks for sharing this feedback and workarounds y'all. I've passed along this request to our Product team. If/when I have any updates - I'll circle back here.

**Reply by chrisglein** [reply]
Lots of related issues on this. To help interlink: #5411 #6217 #47733 #48389 #31983

**Comment by rileybroughten** [comment, marked as answer]
You can now export your project view to a CSV file! Check it out on the [changelog](https://github.blog/changelog/2023-06-15-github-issues-projects-june-15th-update/#%F0%9F%8F%97%EF%B8%8F-export-project-view-as-a-csv-file) announcement and let us know what you think 🚀

**Reply by OscarSal** [reply]
Hi @rileybroughten ,
it will be aweseome (and totally necessary) if this export also include issue ID and the order in the view:

![image](https://github.com/community/community/assets/100857600/3c4c0cd8-5c43-4b07-88fe-03c3eb0fa6ce)

**Reply by sammaniamsam** [reply]
Thank you so much for adding this feature set. This makes the task of analyzing Git issues and Git issue attributes for a given milestone simple.

**Reply by AyanSinhaMahapatra** [reply]
Thanks for adding this!

**Reply by letusskate** [reply]
Thank you!
You are a good team!
Wish github all the best!

**Reply by tiarakelly** [reply]
Hi, Will descriptions be added to the exports in the future?

**Reply by Thuliomac** [reply]
meu caro meu vs code travou nao abre foto nao faz nada estou perdendo a chance de fazer meu primeiro projeto e começar meu curso so isso e ainda nao consegui com o git nao e culpa de vcs mais da outra forma tava mais facil

**Reply by montana64** [reply]
http://blkchairbknpn73cfjhevhla7rkp4ed5gg2knctvv7it4lioy22defid.onion/

**Reply by m1rm** [reply]
> You can now export your project view to a CSV file! Check it out on the [changelog](https://github.blog/changelog/2023-06-15-github-issues-projects-june-15th-update/#%F0%9F%8F%97%EF%B8%8F-export-project-view-as-a-csv-file) announcement and let us know what you think 🚀

Love to have this. We use GH projects to organize our work and we recently ran into what I would call "expected but unfortunate behavior". We use the "**export view data" functionality:**

<img width="451" height="289" alt="image" src="https://github.com/user-attachments/assets/658f325b-9f2d-4752-abcf-0a330a927b1f" />


Which spawns this **confirmation dialog** after clicking:

<img width="320" height="186" alt="exportViewData" src="https://github.com/user-attachments/assets/a124bc77-4339-4616-bf12-d1883cfd43d8" />

We noticed that "rows that have been loaded into this view" but initially misunderstood it. We thought, once we filtered in our project view and the browser tab was done loading, that everything would be included in the export.

What actually happened was, that data was missing in the export because we first had to scroll down in the view and saw issues being loaded when they came into view.

After scrolling down once and exporting again, all data was included in the export.

This might be nice to know for others and as a developer I can somewhat guess why it is that way (data being loaded in the view, the export only containing what's in the state at the time the export button is clicked etc.) but it would be nice to have a way to export all data of a view without having to scroll first.

**Reply by Birendra1245** [reply]
This is a great addition—being able to export project views is genuinely useful for teams relying on GitHub Projects for tracking and reporting.

One thing we ran into, though, might catch others off guard. The export only includes rows that have actually been loaded in the view at the time of export. Initially, we assumed that once the filtered view finished loading, everything would be included—but that’s not the case. Data that hadn’t been brought into view (via scrolling) was missing from the CSV.

After scrolling through the full list and triggering all items to load, the export worked as expected.

Totally understand why it behaves this way from a technical standpoint, but it would be really helpful to have an option to export the complete dataset for a view without needing to manually scroll 

> _first.****_

**Comment by anonimous-maker** [comment]
help im being attacked by a hacker that also broke into github using wikipedia and other methods and stole identity mine and yours github.com

**Comment by buonoi** [comment]
> ### Phản hồi sản phẩm
> Khi tôi sử dụng các dự án github, tôi có cảm giác mượt mà và tự nhiên. Đây thực sự là một điều tốt để tôi quản lý kế hoạch mã hóa của mình. Nhưng khi tôi muốn xuất nó để tạo hình ảnh hoặc powerpoint, tôi thấy thực sự khó khăn. Tôi phải sao chép và dán trong khoảng một ngày. Tôi nghĩ nếu có một biểu tượng để xuất dự án sang csv, thì nó có ý nghĩa.

**Comment by juan430** [comment]
👍

**Comment by Armanidashh** [comment]
Where can I go for help? With a real agent

**Comment by Thuliomac** [comment]
como exporto um projeto csv se nao ta codando nem no vscode mudou tudo as extensoes um monte de coisas que estava acostumado a codar nao tem mais

**Comment by jan454060-ai** [comment]
ff

**Comment by abigonah** [comment]
Hello everyone 
My name is Arghavan Last name is Bigonah I'm very Glad to see everybody here I'm pianist. I'm engineer computer From Iran and student at Law school

**Comment by abigonah** [comment]
I'm Happy to see everybody Here Good luck.

**Comment by mickjigar** [comment]
Can we get a way to export the "Description" (body text) or "comments"?

**Comment by kindera9699-spec** [comment]
Hacked

**Comment by warasingh** [comment]
มีข้อมูลเก่าเพียงแค่นี้เองเหรอควรปิดเว็บไซต์ไปซะและจัดการสำนักพัฒนาที่มีส่วนเกี่ยวข้องกับเว็บไซต์นี้ทุกคนข้อมูลของผมไม่เห็นมีเลย

**Comment by saeydsiso5522-arch** [comment]
انا محمود المعرص الاقرع واحمد ممتاز المتناك واحمد عبد الله الخول وجمال الةسه وopy and paste for about a whole day. I think if there is a icon to export the project to csv, it does make sence.

**Comment by saeydsiso5522-arch** [comment]
محمود العرص الاقرعوبيحب يتناك فى التوته وامهوبتعمل كده اودامه وابويه كان رتجل علق

**Comment by bandar80s** [comment]
function processCommand(command) {
try {
if (!command?.trim()) {
return 'Please enter a command. Type "more" for available commands.';
}

const sanitizedCommand = command.toLowerCase().trim();

switch (sanitizedCommand) {
case 'more':
return commands.more;
case 'about':
return commands.about;
case 'skills':
return commands.skills;
case 'projects':
return commands.projects;
case 'contact':
return commands.contact;
default:
return `Command not found: "${command}". Type 'more' for available commands.`;
}
} catch (error) {
console.error('[Terminal] Error processing command:', error);
return 'An error occurred while processing the command.';
}
}

**Comment by bandar80s** [comment]
> function processCommand(command) {
> try {
> if (!command?.trim()) {
> return 'Please enter a command. Type "more" for available commands.';
> }
> 
> const sanitizedCommand = command.toLowerCase().trim();
> 
> switch (sanitizedCommand) {
> case 'more':
> return commands.more;
> case 'about':
> return commands.about;
> case 'skills':
> return commands.skills;
> case 'projects':
> return commands.projects;
> case 'contact':
> return commands.contact;
> default:
> return `Command not found: "${command}". Type 'more' for available commands.`;
> }
> } catch (error) {
> console.error('[Terminal] Error processing command:', error);
> return 'An error occurred while processing the command.';
> }
> }

**Reply by Adam9911** [reply]
@microsoft

**Comment by Adam9911** [comment]
@microsoft

**Comment by Birendra1245** [comment]
Looks like you want a clean, ready-to-use reply for a “comment / answer” slot. Since there’s no specific question attached, here are a few versatile options you can drop in depending on tone:

Neutral & safe:

> Thanks for raising this. From what I understand, the key point here is balancing practicality with long-term impact. A thoughtful approach would consider both immediate results and sustainability.



Confident & direct:

> The answer is fairly straightforward: focus on what actually works in practice, not just theory. Prioritize efficiency, measure results, and adjust based on real outcomes.



Slightly conversational:

> Good question. Honestly, it comes down to keeping things simple and consistent—overcomplicating it usually creates more problems than it solves.



Supportive / agreeable:

> I agree with the direction here. It makes sense, especially when you look at the bigger picture and how things tend to play out over time.

**Comment by dweb65355-dot** [comment]
I want to earn money from Github and Maddom and I want to get a Bikash
Nogod Agent SIM card from my house. I can support you as much as I can...
But I could not understand Github and Onk... You are my friend,

**Comment by question429** [comment]
Thanks! I was the original poster. Glad this feature is available n
